src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,802 @@
+
+    global cli_outputs
+
+    try:
+
+        if storage_space_available:
+
+            # CMD command to get helm values in azure arc and converting it to json format
+            command = ["kubectl", "get", "events", "-n", "azure-arc", "--output", "json"]
+
+            # Using subprocess to execute the helm get values command and fetching the output
+            response_kubectl_get_events = Popen(command, stdout=PIPE, stderr=PIPE)
+            output_kubectl_get_events, error_kubectl_get_events = response_kubectl_get_events.communicate()
+
+            if response_kubectl_get_events.returncode != 0:
+                telemetry.set_exception(exception=error_kubectl_get_events.decode("ascii"), fault_type=consts.Kubectl_Get_Events_Failed, summary='Error while doing kubectl get events')
+                logger.warning("Error while doing kubectl get events")
+
+            # Converting output obtained in json format and fetching the clusterconnect-agent feature
+            events_json = json.loads(output_kubectl_get_events)
+
+            event_logs_path = os.path.join(filepath_with_timestamp, "Arc_Agents_Events.txt")
+            with open(event_logs_path, 'w+') as event_log:
+
+                for events in events_json["items"]:
+                        event_log.write(str(events) + "\n")
+
+            return "Passed", storage_space_available
+
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An expection has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Arc_Agents_Events_Logger_Fault_Type, summary="Error occured in arc agents events logger")
+            cli_outputs.append("An expection has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Arc_Agents_Events_Logger_Fault_Type, summary="Error occured in arc agents events logger")
+        cli_outputs.append("An expection has occured while trying to fetch the events occured in azure-arc namespace from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return "Failed", storage_space_available
+
+
+def deployments_logger(appv1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    global cli_outputs
+
+    try:
```

Comment:
I see that most of these functions in troubleshootutils.py correspond to a check that we're doing on the cluster. Could we include a log at the beginning of each function that says something like "retrieving deployment logs in azure-arc ns"?

Reply:
Hey sophie, i think we should not add this line because in future the checks may increase and printing each time that we are performing this check may not be a great user experience. User would be confused and would face little difficulty to spot the error. Currently we focus completely on the error.

Reply:
@anraghun to provide any suggestion for above